src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -137,17 +150,18 @@ def get(self, request: Request, project) -> Response:
         )
         matched_code_mappings = []
         with configure_scope() as scope:
-            scope.set_tag("project.slug", project.slug)
-            scope.set_tag("organization.slug", project.organization.slug)
+            set_top_tags(scope, project)
             for config in configs:
-                if not filepath.startswith(config.stack_root) and not mobile_frame:
```

Comment:
`mobile_frame` was **always** set, thus, it would try *all* code mappings even if they did not match the stack trace root (not the right behaviour).

As reference, this is the original code before my last refactoring (related PR #35433):
https://github.com/getsentry/sentry/commit/643701f2d38994fc72631a43e3b339ac8e67f940#diff-9ed2a929b59278cbb7d160ac361180e90dbb3ec7359f7543b614e12e1615d059R102